script.js

```diff
@@ -1,12 +1,275 @@
+    episode.season,
+    episode.number
+  )}`;
+  return title;
+}
+
+function formatEpisodeNum(season, number) {
+  return `S${season.toString().padStart(2, "0")}E${number
+    .toString()
+    .padStart(2, "0")}`;
+}
+
+// ------------- create episode and show Page  ---------------
+
+function createSummary(summary) {
+  let summaryElement = document.createElement("p");
+  summaryElement.style.margin = "30px 30px 20px 30px";
+  summaryElement.style.textAlign = "justify";
+  summaryElement.innerHTML = summary;
+  return summaryElement;
+}
+
+function createImage(imageSrc) {
+  let imageElement = document.createElement("img");
+  imageElement.src = imageSrc;
+  return imageElement;
+}
+
+// ------------- Create Episode Dropdown on onload ------------
+
+function createDropdown(allEpisodes) {
+  episodeSelector.innerHTML = '';
+  createSelectAllEpisodesOption()
+  for (let i = 0; i < allEpisodes.length; i++) {
+    let optionTitle = document.createElement("option");
+    episodeSelector.appendChild(optionTitle);
+    optionTitle.innerText = `${formatEpisodeNum(
+      allEpisodes[i].season,
+      allEpisodes[i].number
+    )} - ${allEpisodes[i].name}`;
+  }
+}
+
+function createSelectAllEpisodesOption() {
+  let optionTitle = document.createElement("option");
+  episodeSelector.appendChild(optionTitle);
+  optionTitle.innerText = "Select All Episodes";
+  
+
+   optionTitle.addEventListener("change", setup);
+}
+
+// ------------ Episode Selector -----------------
+
+episodeSelector.addEventListener("change", function (event) {  
+ 
+  let episodeOption = event.target.value;
+  episodeOption = episodeOption.slice(0, 6);
+  let episodes = document.querySelectorAll(".card");
+
+  let newArrayOfEpisodes = Array.from(episodes);
+  newArrayOfEpisodes.forEach(function (card) {
+    if (!card.innerText.includes(episodeOption)) {
+      card.style.display = "none";
+    } else if (card.innerText.includes(episodeOption)) {
+      card.style.display = "block";
+    }
+  });
+  let filteredListOfEpisodes = newArrayOfEpisodes.filter(
+    (item) => item.style.display === "block"
+  );
+  displayNumOfEpisodes(filteredListOfEpisodes.length, newArrayOfEpisodes.length);
+});
+
+// ------------- Search bar  --------------------- 
+
+input.addEventListener("keyup", function (e) {
+  const term = e.target.value.toLowerCase();
+  let episodes = document.querySelectorAll(".card");
+  let newArrayOfEpisodes = Array.from(episodes);
+  newArrayOfEpisodes.forEach(function (episode) {
+    if (episode.innerText.toLowerCase().includes(term)) {
+      episode.style.display = "block";
+    } else {
+      episode.style.display = "none";
+    }
+  });
+  let filteredListOfEpisodes = newArrayOfEpisodes.filter(
+    (item) => item.style.display === "block"
+  );
+  if (newArrayOfEpisodes[0].id != "") {
+    displayNumOfShows(filteredListOfEpisodes.length, newArrayOfEpisodes.length);
+  }
+  displayNumOfEpisodes(filteredListOfEpisodes.length, newArrayOfEpisodes.length);
```

Comment:
You need to put `displayNumOfEpisodes` in an `else` block because it runs both functions when if condition is true and there is an error in the console.